src/app/InteractionModelEngine.h

```diff
@@ -698,7 +705,23 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
 
     SubscriptionResumptionStorage * mpSubscriptionResumptionStorage = nullptr;
 
-    DataModel::Provider * mDataModelProvider = nullptr;
+    DataModel::Provider * mDataModelProvider      = nullptr;
+    Messaging::ExchangeContext * mCurrentExchange = nullptr;
+
+    // Changes the current exchange context of a InteractionModelEngine to a given context
```

Comment:
It's funny that we went to all the trouble of getting rid of this sort of global state and just passing relevant context to command handlers as part of CommandHandler, and now we are walking it back....

Put another way, maybe the ActionContext for an action should not be a global?  Followup to think about that.

Reply:
This is because we started using global exchange manager in places. Very few places, however without it we cannot say the new code supports all the functionality of the old code.

I would prefer we would not expose this and find some other way to process things. Even comments on the context way "use sparringly, likely null in tests because it is impractical to initialize such a complex object as a mock".